azurerm/internal/services/avs/avs_private_cloud_resource.go

```diff
@@ -0,0 +1,413 @@
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Avs Private Cloud %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("retrieving Avs PrivateCloud %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+	d.Set("name", id.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("location", location.NormalizeNilable(resp.Location))
+	if props := resp.PrivateCloudProperties; props != nil {
+		if err := d.Set("management_cluster", flattenArmPrivateCloudManagementCluster(props.ManagementCluster)); err != nil {
+			return fmt.Errorf("setting `management_cluster`: %+v", err)
+		}
+		d.Set("network_subnet", props.NetworkBlock)
+		if err := d.Set("circuit", flattenArmPrivateCloudCircuit(props.Circuit)); err != nil {
+			return fmt.Errorf("setting `circuit`: %+v", err)
+		}
+
+		d.Set("internet_connection_enabled", props.Internet == avs.Enabled)
+		d.Set("hcx_cloud_manager_endpoint", props.Endpoints.HcxCloudManager)
+		d.Set("nsxt_manager_endpoint", props.Endpoints.NsxtManager)
+		d.Set("vcsa_endpoint", props.Endpoints.Vcsa)
+		d.Set("management_subnet", props.ManagementNetwork)
+		d.Set("nsxt_certificate_thumbprint", props.NsxtCertificateThumbprint)
+		d.Set("provisioning_subnet", props.ProvisioningNetwork)
+		d.Set("vcenter_certificate_thumbprint", props.VcenterCertificateThumbprint)
+		d.Set("vmotion_subnet", props.VmotionNetwork)
+	}
+	d.Set("sku_name", resp.Sku.Name)
+	return tags.FlattenAndSet(d, resp.Tags)
+}
+
+func resourceAvsPrivateCloudUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Avs.PrivateCloudClient
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.PrivateCloudID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	privateCloudUpdate := avs.PrivateCloudUpdate{
+		PrivateCloudUpdateProperties: &avs.PrivateCloudUpdateProperties{},
+	}
+	if d.HasChange("management_cluster") {
+		privateCloudUpdate.PrivateCloudUpdateProperties.ManagementCluster = &avs.ManagementCluster{
+			ClusterSize: utils.Int32(int32(d.Get("management_cluster.0.size").(int))),
+		}
+	}
+	if d.HasChange("internet_connection_enabled") {
+		internet := avs.Disabled
+		if d.Get("internet_connection_enabled").(bool) {
+			internet = avs.Enabled
+		}
+		privateCloudUpdate.PrivateCloudUpdateProperties.Internet = internet
+	}
+
+	if d.HasChange("tags") {
+		privateCloudUpdate.Tags = tags.Expand(d.Get("tags").(map[string]interface{}))
+	}
+
+	future, err := client.Update(ctx, id.ResourceGroup, id.Name, privateCloudUpdate)
+	if err != nil {
+		return fmt.Errorf("updating Avs PrivateCloud %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting on updating future for Avs PrivateCloud %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
```

Comment:
(as above)

we also don't mention `Future`'s externally, they're an implementation detail - this wants to be:

```suggestion
		return fmt.Errorf("waiting for update of VMware Private Cloud %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
```